src/Docker/Docker.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace DockerHostManager\Docker;
+
+use Docker\Docker as DockerBase;
+
+class Docker extends DockerBase
+{
+    /**
+     * @param string   $entrypoint
+     * @param callable $callback
+     */
+    public function listenEvents($entrypoint, callable $callback)
+    {
+        $stream = stream_socket_client($entrypoint, $errno, $errstr, 30);
```

Comment:
Did you try to use `DockerAdapter::send()` instead of creating your own stream ?